exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -792,124 +1294,328 @@ func (e *Exchange) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBui
+					continue
+				}
+				oSide, err = order.StringToOrderSide(tOrder.Side)
+				if err != nil {
+					return nil, err
+				}
+				oType, err = order.StringToOrderType(tOrder.Type)
+				if err != nil {
+					return nil, err
+				}
+				var assetType asset.Item
+				assetType, err = asset.New(tOrder.AccountType)
+				if err != nil {
+					return nil, err
+				}
+				detail := order.Detail{
+					Side:                 oSide,
+					Amount:               tOrder.Amount.Float64(),
+					ExecutedAmount:       tOrder.FilledAmount.Float64(),
+					Price:                tOrder.Price.Float64(),
+					AverageExecutedPrice: tOrder.AveragePrice.Float64(),
+					Pair:                 cp,
+					Type:                 oType,
+					Exchange:             e.Name,
+					QuoteAmount:          tOrder.Amount.Float64() * tOrder.AveragePrice.Float64(),
+					RemainingAmount:      tOrder.Quantity.Float64() - tOrder.FilledQuantity.Float64(),
+					OrderID:              tOrder.ID,
+					ClientOrderID:        tOrder.ClientOrderID,
+					Status:               orderStateFromString(tOrder.State),
+					AssetType:            assetType,
+					Date:                 tOrder.CreateTime.Time(),
+					LastUpdated:          tOrder.UpdateTime.Time(),
+					TimeInForce:          tOrder.TimeInForce,
+				}
+				detail.InferCostsAndTimes()
+				orders = append(orders, detail)
+			}
+			return req.Filter(e.Name, orders), nil
+		case order.Stop, order.StopLimit, order.TrailingStop, order.TrailingStopLimit:
+			oTypeString, err := orderTypeString(req.Type)
+			if err != nil {
+				return nil, err
+			}
+			smartOrders, err := e.GetSmartOrderHistory(ctx,
+				&OrdersHistoryRequest{
+					Symbol:      currency.EMPTYPAIR,
+					AccountType: accountTypeString(req.AssetType),
+					OrderType:   oTypeString,
+					Side:        req.Side,
+					Limit:       100,
+					StartTime:   req.StartTime,
+					EndTime:     req.EndTime,
+				},
+			)
+			if err != nil {
+				return nil, err
+			}
+			var (
+				oSide order.Side
+				oType order.Type
+			)
+			orders := make([]order.Detail, 0, len(smartOrders))
+			for _, smartOrder := range smartOrders {
+				if len(req.Pairs) != 0 && !req.Pairs.Contains(smartOrder.Symbol, true) {
+					continue
+				}
+				oSide, err = order.StringToOrderSide(smartOrder.Side)
+				if err != nil {
+					return nil, err
+				}
+				oType, err = order.StringToOrderType(smartOrder.Type)
+				if err != nil {
+					return nil, err
+				}
+				assetType, err := asset.New(smartOrder.AccountType)
+				if err != nil {
+					return nil, err
+				}
+				detail := order.Detail{
+					Side:          oSide,
+					Amount:        smartOrder.Amount.Float64(),
+					Price:         smartOrder.Price.Float64(),
+					TriggerPrice:  smartOrder.StopPrice.Float64(),
+					Pair:          smartOrder.Symbol,
+					Type:          oType,
+					Exchange:      e.Name,
+					OrderID:       smartOrder.ID,
+					ClientOrderID: smartOrder.ClientOrderID,
+					Status:        order.Filled,
```

Comment:
Have said before, don't hardcode this: https://api-docs.poloniex.com/spot/api/private/order-history

> FAILED, FILLED, CANCELED. PARTIALLY_CANCELED